include/fcntl.h

```diff
@@ -15,8 +15,13 @@
 #define O_DIRECT     040000
 #define O_NOATIME  01000000
 #elif ((defined CONFIG_ARCH_ARM_64) || (defined CONFIG_ARCH_ARM_32))
+// redefine FNONBLOCK when O_NONBLOCK is redefined
+#undef FNONBLOCK
```

Comment:
This should actually happen for all architectures. [This commit](https://github.com/unikraft/lib-newlib/commit/58f9f3a3f7b5ca9c5f0c7c095e067abba29f5ed6) made the change for ARM, but it should happen for all architectures. `nolibc` defines the same value for both [x86_64](https://github.com/unikraft/unikraft/blob/staging/lib/nolibc/arch/x86_64/bits/fcntl.h) and for [ARM](https://github.com/unikraft/unikraft/blob/staging/lib/nolibc/arch/arm64/bits/fcntl.h). So, the two lines above should go both for x86_64 and for ARM. I think it's cleanest to duplicate them for each architectural `#ifdef`, connected to the other macros:

```
#undef FNONBLOCK
#undef O_NONBLOCK
#define FNONBLOCK 04000
#define O_NONBLOCK FNONBLOCK
```

I validated this working on my side.